src/graph/graph.c

```diff
@@ -55,11 +72,16 @@ void _Graph_ResizeMatrix(const Graph *g, GrB_Matrix m) {
         {
             // Double check now that we're in critical section.
             GrB_Matrix_nrows(&n_rows, m);
-            if(n_rows != g->node_count)
+            if(n_rows != g->node_count) {
                 assert(GxB_Matrix_resize(m, g->node_count, g->node_count) == GrB_SUCCESS);
+                // TODO this step is still required if the matrix ever had an explicit 0 added,
+                // regardless of whether it currently has the correct node count.
+                _Graph_ReduceMatrix(m);
```

Comment:
This is where I'm concerned that the conditional might not be sufficient. The dangerous situation would be:
- 1 or more nodes with label _x_ are deleted, and their IDs are replaced by nodes without that label.
- The same number of nodes are added, none of which have label _x_
- A label scan for _x_ is performed.

In this scenario, the conditional at 75 is not met, but we still have explicit 0 values that will be collected by the `TuplesIter`.